plaid-stl/src/npm/mod.rs

```diff
@@ -0,0 +1,607 @@
+use serde::{Deserialize, Deserializer, Serialize};
+
+use crate::PlaidFunctionError;
+
+use chrono::{DateTime, Utc};
+
+// All the structs which are also used in the Plaid API
+
+/// Permission that is granted to a team over an npm package
+#[derive(Serialize, Deserialize, Clone)]
+pub enum NpmPackagePermission {
+    READ,
+    WRITE,
+}
+
+impl ToString for NpmPackagePermission {
+    fn to_string(&self) -> String {
+        match self {
+            Self::READ => "read".to_string(),
+            Self::WRITE => "write".to_string(),
+        }
+    }
+}
+
+impl NpmPackagePermission {
+    // We do not implement the TryFrom trait because of the error type. We do not want to carry the
+    // NpmError struct here in the STL. Also, we don't care about which error we get from here. If
+    // it fails, the caller will know what to do.
+    pub fn try_from(value: String) -> Result<Self, ()> {
+        match value.as_str() {
+            "read" => Ok(NpmPackagePermission::READ),
+            "write" => Ok(NpmPackagePermission::WRITE),
+            _ => Err(()),
+        }
+    }
+}
+
+/// Teams in the npm organization
+#[derive(Serialize, Deserialize)]
+pub enum NpmTeam {
```

Comment:
Is this defined by NPM? Can't we have arbitrary teams?

Reply:
This is replicating what we already have configured in the organization. In theory, one can always create new teams from the website and add them here, if they should be managed by the API.
For example, on the website there are two more teams for auditors that I did not expose here.

Reply:
If we want to make everything generic, and indeed have arbitrary teams, we can just take this as a rule-defined string.

The idea of using an enum was to prevent silly mistakes like typos, but the flip side is that we have to know all the possibilities beforehand.

Reply:
Moved to using strings so that rules have more flexibility. Let's see if this works fine.